mantidproject/mantid
# Batch files to not require an output name

**Description of work.**
This PR stops raising a RuntimeError when parsing a batch file if it does not contain an output name.
In this case, an auto-generated name is used instead

**Report to:** Rob D

**To test:**
1. Interfaces -> SANS -> ISIS SANS
2. Load the attached  user and batch file. This should load without issue
3. Click process all
4. After processing, the ADS should contain workspaces with name stems related to the sample transmission run number (57290). These are the autogenerated workspace names used instead of a user provided one

**Data:**
[BatchFilePRData.zip](https://github.com/mantidproject/mantid/files/3028981/BatchFilePRData.zip)

Fixes #25451 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
